packages/core-ethereum/src/ethereum.ts

```diff
@@ -383,11 +370,24 @@ export async function createChainWrapper(providerURI: string, privateKey: Uint8A
     return transaction.hash
   }
 
+  async function getNativeTokenTransactionInBlock(
```

Comment:
not sure I understand the name of this function, from what I see in the code, it returns transactions to / from the wallet?

Reply:
It returns transactions to/from the wallet with a `tx.value > 0`, meaning transactions that transfers some native tokens. Not all the transactions that moves native tokens are tracked here (e.g. "interacting with a smart contract and the smart contract transfers some native tokens" is out of scope). This function aims at filtering possible `withdrawal` transactions.